### ISSUE TITLE
Optimisation on dataset creation

### ISSUE THREAD

**Issue by theodu** [mlzxy/devit#14]
I have another question regarding the dataset building that will feed the prototypes of novels classes.
My use case is to annotate new images with already annotated images that I will use to build my prototypes.
What would you suggest as the optimal way to build context image masks ?

For exemple, for this image from the [blood-cell-object-detection](https://huggingface.co/datasets/keremberke/blood-cell-object-detection) (3 classes: rbc, wbc and platelets)
![clnlpt9um00003b6oi53f8m38](https://github.com/mlzxy/devit/assets/82399630/09c2b38f-e3aa-41b8-af6b-e5cb21586844)

I tried many different inputs such as:
- a few instances of each classes like:
![clnlpt9um00003b6oi53f8m38 mask](https://github.com/mlzxy/devit/assets/82399630/3afacc84-481b-441c-b4d9-4e966fdccb98)

- all instances of a class like:
![clnlpt9um00003b6oi53f8m38 mask](https://github.com/mlzxy/devit/assets/82399630/4005b0ff-0d8b-48c3-96b9-8a418adaa1d9)

- only one instance of a class like:
![2clnlpt9um00003b6oi53f8m38 mask](https://github.com/mlzxy/devit/assets/82399630/f1c4731a-094c-4a22-afaa-730058bba72d)

- I didn't try yet to crop the image around its instances to create several subset image with one big object inside

But all of these methods output results that are not accurate enough like:
![test_0 out](https://github.com/mlzxy/devit/assets/82399630/28313c9c-5f95-4630-87f8-1754be4dda7f)

In your YCB demo, you use images with one big object at the center of the image. Is there a way for me to optimise my dataset ?

**Comment by mlzxy** [comment]
I would suggest to finetune the region proposal network on the blood cell dataset.  The current approach is based on existing region proposal networks, which is trained on COCO / LVIS. So it naturally performs better in common life scene instead of specific use cases like biology. 

But honestly I feel like a domain shift this large could be too much for a pretrained model could handle, so I could not guarantee this method gonna work definitely. My suggestion is to finetune, if possible, first RPN, then the RCNN if still not sufficient.

**Comment by theodu** [comment]
Thank you for your answer!

**Comment by theodu** [comment]
I will indeed try to retrain the RPNN and/or the RCNN. 
And after that, what would you suggest as the best mask approach for the dataset ?

**Comment by mlzxy** [comment]
I think the best mask approach depends on your application requirement, like whether a single cell shall be detect separately may need to be determined by the actual use case and users (biologist in this case?).

**Comment by gautham-98** [comment]
@theodu Were you able to fine tune the model?
